src/main/kotlin/org/rust/ide/inspections/fixes/AttachFileToModuleFix.kt

```diff
@@ -164,20 +163,28 @@ private fun insertFileToModule(file: RsFile, mod: RsMod) {
     }
 
     WriteCommandAction.runWriteCommandAction(project) {
-        val child = mod.firstChild
-        val inserted = if (child == null) {
-            mod.add(modItem)
-        } else {
-            mod.addBefore(modItem, child)
-        } as RsModDeclItem
-        inserted.navigate(true)
+        insertModItem(modItem, targetFile).navigate(true)
     }
 }
 
+private fun insertModItem(item: RsModDeclItem, module: RsFile): RsModDeclItem {
+    val anchor = module.firstItem
+    val existingMod = (anchor as? RsModDeclItem) ?: anchor?.rightSiblings
```

Comment:
If you have more than one mod declarations in target file, this code inserts new mod declaration after the first one instead of the latest one. Consider using something like `module.childrenOfType<RsModDeclItem>().lastOrNull()`. Yep, it creates an intermediate list object and it's not lazy. But laziness doesn't make sense here because we iterate over all elements anyway and suggested version is more readable, in my opinition
Let's add test with several existent mod declarations to check it

Reply:
Damn, I'm surprised that none of the tests covered this. The cast was in a bad place, it should have been 
```
val existingMod = anchor?.rightSiblings
        ?.filterIsInstance<RsModDeclItem>()
        ?.lastOrNull()
        ?: anchor as? RsModDeclItem
```
My solution skips `mod` items before attributes, but that's an error anyway, so I used yours, it's more readable.